app/controllers/users_controller.rb

```diff
@@ -0,0 +1,37 @@
+class UsersController < ApplicationController
+
+  def new
+    @user = User.new
+  end
+
+  def create
+    user = User.new(user_params)
+
+    if user.save
+      session[:user_id] = user.id
+      redirect_to root_path
+    else
+      flash[:warn] = "User could not be saved"
+      redirect_to new_user_path
+    end
+  end
+
+  def show
+  end
+
+  def edit
+  end
+
+  def update
+  end
+
+  def destroy
```

Comment:
I suspect this one just shouldn't exist - are you allowing users to be deleted? No, there's no code here. So just delete the empty action.